lib/model.js

```diff
@@ -846,6 +863,49 @@ Model.discriminator = function(name, schema) {
   return d;
 };
 
+/**
+ * Check if document implements a Schema
+ *
+ * @param {String|Schema} kind
+ * @return {Boolean}
+ * @api public
+ */
+
+Model.prototype.ofKind = function(kind) {
+  if (kind instanceof Schema) {
+    if (!kind.options.name)
+      return false;
+    kind = kind.options.name;
+  }
+
+  if (!this._kind && (!this._kinds || !Array.isArray(this._kinds)))
+    return false;
+
+  return this._kind === kind || this._kinds.indexOf(kind) > -1;
+}
+
+/**
+ * Check if model implements a Schema
+ *
+ * @param {String|Schema} kind
+ * @return {Boolean}
+ * @api public
+ */
+
+Model.ofKind = function(kind) {
```

Comment:
This seems like it'll be confusing because of discriminators - it'll give us multiple notions of inheritance, which'll inevitably add some weird edge cases